mindmeld/components/question_answerer.py

```diff
@@ -256,23 +299,74 @@ def load_kb(
                 connection to the Elasticsearch host.
             clean (bool): Set to true if you want to delete an existing index
                 and reindex it
+            app_path (str): The path to the directory containing the app's data
+            config (dict): The QA config if passed directly rather than loaded from the app config
         """
+        embedder_model = None
+        embedding_fields = []
+        if not app_path and not config:
+            logger.warning(
+                "You must provide either the application path to upload embeddings as specified"
+                " in the app config or directly provide the QA config."
+            )
+        else:
+            if config:
+                qa_config = config
+            else:
+                qa_config = get_classifier_config(
+                    "question_answering", app_path=app_path
+                )
+            embedder_model = create_embedder_model(app_path, qa_config)
+            embedding_fields = (
+                qa_config.get("model_settings", {})
+                .get("embedding_fields", {})
+                .get(index_name, [])
+            )
+
+        def _doc_data(data_file, embedding_fields):
+            def match_regex(string, pattern_list):
+                for pattern in pattern_list:
+                    if re.match(pattern, string):
+                        return True
 
-        def _doc_count(data_file):
             with open(data_file) as data_fp:
                 line = data_fp.readline()
                 data_fp.seek(0)
                 if line.strip() == "[":
                     docs = json.load(data_fp)
-                    return len(docs)
+                    count = len(docs)
+                    doc_example = docs[0]
```

Comment:
This seems like something that could surprise the developer. Suppose they have an inventory with up to 3 reviews, with keys like "product_review_0", "product_review_1" ... The first doc might have only one or none of these, and then the fields would get left out of the embeddings.

Reply:
Yeah good point, updating